From 62283a977301933aa88304a62304f632a09c3d7a Mon Sep 17 00:00:00 2001
From: Jiucheng Xu <jiucheng.xu@amlogic.com>
Date: Wed, 14 Dec 2022 12:45:57 +0800
Subject: [PATCH 85/95] md: dm bow for kernel 5.15 [1/2]

PD#SWPL-102756

Problem:
Upgrade project need dm bow for ota

Solution:
Port it from 5.4 code

Verify:
sc2

Change-Id: Ie50e77502c9670bf00d73ea77bd379501036d12f
Signed-off-by: Jiucheng Xu <jiucheng.xu@amlogic.com>
---
 drivers/md/Makefile   | 2 ++
 drivers/md/dm-ioctl.c | 8 ++++++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/md/Makefile b/drivers/md/Makefile
index 7890bcc727f5..5459e3842c81 100644
--- a/drivers/md/Makefile
+++ b/drivers/md/Makefile
@@ -110,3 +110,5 @@ endif
 ifeq ($(CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG),y)
 dm-verity-objs			+= dm-verity-verify-sig.o
 endif
+
+ccflags-y += -I$(srctree)/$(COMMON_DRIVERS_DIR)/drivers/md/
diff --git a/drivers/md/dm-ioctl.c b/drivers/md/dm-ioctl.c
index dcaca4aaac91..e49113979fdb 100644
--- a/drivers/md/dm-ioctl.c
+++ b/drivers/md/dm-ioctl.c
@@ -23,6 +23,10 @@
 #include <linux/uaccess.h>
 #include <linux/ima.h>
 
+#ifdef CONFIG_AMLOGIC_DM_BOW
+#include "dm-bow.h"
+#endif
+
 #define DM_MSG_PREFIX "ioctl"
 #define DM_DRIVER_EMAIL "dm-devel@redhat.com"
 
@@ -1465,6 +1469,10 @@ static int table_load(struct file *filp, struct dm_ioctl *param, size_t param_si
 		goto err_unlock_md_type;
 	}
 
+#ifdef CONFIG_AMLOGIC_DM_BOW
+	bow_sysfs_create(t, param, param_size);
+#endif
+
 	dm_unlock_md_type(md);
 
 	/* stage inactive table */
-- 
2.25.1

